crates/bevy_ui/src/geometry.rs

```diff
@@ -45,6 +55,70 @@ pub enum Val {
     VMax(f32),
 }
 
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum ValParseError {
+    UnitMissing,
+    ValueMissing,
+    InvalidValue,
+    InvalidUnit,
+}
+
+impl core::fmt::Display for ValParseError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            ValParseError::UnitMissing => write!(f, "unit missing"),
+            ValParseError::ValueMissing => write!(f, "value missing"),
+            ValParseError::InvalidValue => write!(f, "invalid value"),
+            ValParseError::InvalidUnit => write!(f, "invalid unit"),
+        }
+    }
+}
+
+impl core::str::FromStr for Val {
+    type Err = ValParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
Would love to see this pulled out as a `const` function if it's compatible. Not a blocker if it's not a simple change though.

Reply:
I checked and every single function that's being called is non-const :laughing:

Reply:
Specifically: `str::trim`, `str::bytes()`, `iterator::position`, `str::split_at`, `?`, `f32::parse`

Reply:
Damn, thanks for clarifying!